### ISSUE TITLE
Doesn't work with pure functions as parameter

### ISSUE THREAD

**Issue by dmdque** [sc-forks/solidity-coverage#263]
I'm using a [library function](https://github.com/cryptofinlabs/cryptofin-solidity/blob/master/contracts/array-utils/UIntArrayUtils.sol#L6) which consumes a pure function as a parameter:

```
library UIntArrayUtils {

  function map(uint256[] memory A, function(uint256) pure returns (uint256) fn)
    internal
    pure
    returns (uint256[] memory)
  {
    uint256[] memory mapped = new uint256[](A.length);
    for (uint256 i = 0; i < A.length; i++) {
      mapped[i] = fn(A[i]);
    }
    return mapped;
  }

...
```

And get the following:

```
TypeError: Invalid type for argument in function call. Invalid implicit conversion from function (uint256) pure returns (uint256) to function (uint256) returns (uint256) requested.
```

It looks like [`preprocessor.run`](https://github.com/sc-forks/solidity-coverage/blob/master/lib/preprocessor.js#L69) only considers function declarations, not functions as an argument. Seems like a good place to start investigating.

**Comment by cgewecke** [comment]
@dmdque Interesting! There's also an issue testing pure functions in Libraries that's documented in #234 (there's a work around described). Will look into this, thanks for the nice example.

**Comment by dmdque** [comment]
Got it, don't think I've run into that one.

My current temporary workaround is to modify all affected functions to be non-pure (and non-view).

**Comment by cgewecke** [comment]
Fixed in `0.7.0`
